Chatterino/chatterino2
# Fix tooltips appearing too large and/or away from the cursor

# Description

How to reproduce:

0. Enable link previews.
1. Send `anyway https://clips.twitch.tv/HeartlessShinyJuiceMingLee and https://livestreamfails.com/post/30495` in chat.
2. Resize the window, so both links appear above each other.
3. Move the mouse between the links.
4. Notice the tooltip appearing really large for a frame or two (this often happens when hovering really fast between the links).
5. Notice the tooltip not appearing above the cursor (this often happens when hovering slow).

---

This PR fixes the issue by recalculating the size of the widgets if it's unexpected and refreshing when an image gets loaded (rather than every 100ms). Furthermore, it avoids any work if the window isn't visible.
